es-core/src/PowerSaver.cpp

```diff
@@ -46,6 +46,7 @@ void PowerSaver::updateMode()
 	} else {
 		mMode = DEFAULT;
 	}
+	updateTimeouts();
```

Comment:
UpdateTimeouts is in updateMode() and not getmode(). I place it these as update timeouts should always be called after setting mode and always in that order. Initially i had it other way around and noticed that it was always returning 60001 and 30001 ie `value - (-1)` where -1 was unset value. hence the change.
***
Since get mode is only used as a comparator to check if in different modes i didnt bother changing but i think that another method doing the exact same thing can be done if you think so:

```
// The only place it in used as timeout
es-app/src/main.cpp: bool ps_standby = PowerSaver::getState() && SDL_GetTicks() - ps_time > PowerSaver::getMode();
es-app/src/SystemScreenSaver.cpp: if (mState == STATE_SCREENSAVER_ACTIVE || PowerSaver::getMode() == PowerSaver::INSTANT)
es-app/src/guis/GuiMenu.cpp:	 && PowerSaver::getMode() == PowerSaver::INSTANT)
es-app/src/guis/GuiMenu.cpp:	 && PowerSaver::getMode() == PowerSaver::INSTANT)

// Internal usage
es-core/src/PowerSaver.cpp:	mScreenSaverTimeout = mScreenSaverTimeout > 0 ? mScreenSaverTimeout - getMode() : -1;
es-core/src/PowerSaver.cpp:	mPlayNextTimeout = 30000 - getMode();
es-core/src/PowerSaver.cpp:PowerSaver::mode PowerSaver::getMode()


```